bundles/org.openhab.binding.lghombot/src/main/java/org/openhab/binding/lghombot/internal/LGHomBotHandler.java

```diff
@@ -0,0 +1,722 @@
+                    case "JSON_ROBOT_STATE":
+                        if (value.isEmpty()) {
+                            value = HBSTATE_UNKNOWN;
+                        }
+                        if (!value.equals(currentState)) {
+                            currentState = value;
+                            channel = new ChannelUID(getThing().getUID(), CHANNEL_STATE);
+                            updateState(channel, StringType.valueOf(value));
+
+                            if (value.equals(HBSTATE_WORKING) || value.equals(HBSTATE_BACKMOVING)
+                                    || value.equals(HBSTATE_BACKMOVING_INIT)) {
+                                currentCleanState = OnOffType.ON;
+                                currentStartState = OnOffType.ON;
+                                currentHomeState = OnOffType.OFF;
+                            } else if (value.equals(HBSTATE_HOMING) || value.equals(HBSTATE_DOCKING)) {
+                                currentCleanState = OnOffType.OFF;
+                                currentStartState = OnOffType.OFF;
+                                currentHomeState = OnOffType.ON;
+                            } else {
+                                currentCleanState = OnOffType.OFF;
+                                currentStartState = OnOffType.OFF;
+                                currentHomeState = OnOffType.OFF;
+                            }
+                            channel = new ChannelUID(getThing().getUID(), CHANNEL_CLEAN);
+                            updateState(channel, currentCleanState);
+                            channel = new ChannelUID(getThing().getUID(), CHANNEL_START);
+                            updateState(channel, currentStartState);
+                            channel = new ChannelUID(getThing().getUID(), CHANNEL_HOME);
+                            updateState(channel, currentHomeState);
+                        }
+                        break;
+                    case "JSON_BATTPERC":
+                        try {
+                            DecimalType battery = DecimalType.valueOf(value);
+                            if (!battery.equals(currentBattery)) {
+                                currentBattery = battery;
+                                channel = new ChannelUID(getThing().getUID(), CHANNEL_BATTERY);
+                                updateState(channel, battery);
+                            }
+                        } catch (NumberFormatException e) {
+                            logger.debug("Couldn't parse Battery Percent.");
+                            parsingOk = false;
+                        }
+                        break;
+                    case "CPU_IDLE":
+                        if (isLinked(CHANNEL_CPU_LOAD)) {
+                            try {
+                                DecimalType cpuLoad = new DecimalType(100 - Double.valueOf(value).longValue());
+                                if (!cpuLoad.equals(currentCPULoad)) {
```

Comment:
AFAIK there is no reason to have this check. And please split this of in a method.